xarray/tests/test_backends.py

```diff
@@ -1509,6 +1509,22 @@ def test_cmp_local_file(self):
             self.assertDatasetEqual(actual.isel(j=slice(1, 2)),
                                     expected.isel(j=slice(1, 2)))
 
+    def test_password(self):
+        from pydap.cas.urs import setup_session
+
+        url = ('https://disc2.gesdisc.eosdis.nasa.gov/opendap/TRMM_RT/'
```

Comment:
I agree completely with regards to the other pydap tests being sufficient for integration testing, and also feel that hard-coding a URL like this is bad practice. However i wanted to show how/that i tested the implementation.

I'm not sure about the server staying online at that address indefinitely. Then there's also the problem of the credentials being publicly available here, so people may use it and change the password.

Good idea mocking the function! I also feel confident about the correctness of this, but i'd like to give mocking a try, since i don't really have any experience with that.